lib/mcp/transports/rack_transport.rb

```diff
@@ -384,7 +399,10 @@ def setup_sse_connection(client_id, io, env)
 
         # Send endpoint information as the first message with query parameters
         endpoint = "#{@path_prefix}/#{@messages_route}"
-        endpoint += "?#{query_string}" if query_string
+        params = []
+        params << query_string if query_string && !query_string.empty?
+        params << "client_id=#{client_id}"
+        endpoint += "?#{params.join('&')}" unless params.empty?
```

Comment:
params will always contain at least the client_id and will never be empty then 👍 
```suggestion
        endpoint += "?#{params.join('&')}"
```